### PR TITLE
Removing domain sync threads where not currently used.

### PR DESCRIPTION
**Pull request by lepsa**
- Remove the domain sync thread from galley and gundeck.
- Remove the brig config value from gundeck.

## Checklist

 - [:heavy_check_mark:] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [:heavy_check_mark:] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: libs/wire-api/src/Wire/API/FederationUpdate.hs [hunk 1]
```diff
@@ -32,9 +32,9 @@ syncFedDomainConfigs (Endpoint h p) log' cb = do
 -- | Initial function for getting the set of domains from brig, and an update interval
 initialize :: L.Logger -> ClientEnv -> IO FederationDomainConfigs
 initialize logger clientEnv =
-  let -- keep trying every 3s for one minute
+  let -- An initial value of 100000 is used throughout services. Another commonly used value is 50000
       policy :: R.RetryPolicy
-      policy = R.constantDelay 3_081_003 <> R.limitRetries 20
+      policy = R.exponentialBackoff 100000 <> R.limitRetries 20
```
Comment (akshaymankar) [inline libs/wire-api/src/Wire/API/FederationUpdate.hs:37]: I think we should always add a cap to the exponential backoff, so we don't get crazy wait times like 1 hour.

Comment (lepsa) [reply libs/wire-api/src/Wire/API/FederationUpdate.hs:37]: Luckily Matthias has a PR for exactly that! https://github.com/wireapp/wire-server/pull/3401

File: changelog.d/0-release-notes/domain-sync-thread-cleanup [hunk 2]
```diff
@@ -0,0 +1 @@
+Removed brig configuration value from gundeck.
```
Comment (fisx) [inline changelog.d/0-release-notes/domain-sync-thread-cleanup:1]:
```suggestion
Removed federation remote sync look from gundeck and galley
```

File: libs/wire-api/src/Wire/API/FederationUpdate.hs (hunk 1 above)
Comment (fisx) [reply libs/wire-api/src/Wire/API/FederationUpdate.hs]: oh, and resolving the conflict made me notice i forgot to remove the comment.  :-)  this mixup was a success!

**Review by fisx — APPROVED** [review]

**Pull request merged**